src/deque.py

```
@@ -0,0 +1,67 @@
+"""Implementation of Deque."""
+
+from dll import DoublyLinkedList
+
+
+class Deque(object):
+    """Class implementation of queue.
+
+    append(val): adds value to the end of the deque.
+    appendleft(val): adds a value to the front of the deque.
+    pop(): removes a value from the end of the deque and returns it
+            (raises an exception if the deque is empty)
+    popleft(): removes a value from the front of the deque and returns it
+            (raises an exception if the deque is empty)
+    peek(): returns the next value that would be returned by pop but
+            leaves the value in the deque (returns None if the deque is empty)
+    peekleft(): returns the next value that would be returned by popleft
+            but leaves the value in the deque (returns None if the deque
+            is empty)
+    size(): returns the count of items in the queue (returns 0 if the
+            queue is empty)
+
+    """
+
+    def __init__(self, iterable=None):
+        """Instatiate Queue."""
+        self.dll = DoublyLinkedList(iterable)
+
+    def append(self, contents):
+        """Add value to the end of the deque."""
+        self.dll.append(contents)
```

Something isn't hooked up right under the hood...

```python
In [3]: dq = Deque()

In [4]: dq.dll.head_node

In [5]: dq.dll.tail_node

In [6]: dq.append(1)

In [7]: dq.append(2)

In [8]: dq.dll.head_node.next_node

In [9]: dq.dll.tail_node.next_node

In [10]: dq.dll.tail_node.prev_node
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-10-168a0eb6c06d> in <module>()
----> 1 dq.dll.tail_node.prev_node

AttributeError: 'Node' object has no attribute 'prev_node'

In [11]: dq.dll.head_node.prev_node
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-11-5af9e8cfb3ed> in <module>()
----> 1 dq.dll.head_node.prev_node

AttributeError: 'Node' object has no attribute 'prev_node'
```

---

Either head or tail should have a next node, the other should have a prev.  Neither should throw an error once at least one item has been added (should instead should return none).